python/triton/experimental/gluon/language/amd/_layouts.py

```diff
@@ -65,9 +69,69 @@ def _to_ir(self, builder):
 
     def mangle(self) -> str:
 
-        def stringify(x):
-            if x is None:
-                return ""
-            return "_".join(map(str, x))
+        return f"MFMA_{self.version}_{_stringify(self.instr_shape)}_{self.transposed}_{_stringify(self.warps_per_cta)}_{_stringify(self.tiles_per_warp)}_{self.elem_type}_{_stringify(self.ctas_per_cga)}_{_stringify(self.cta_split_num)}_{_stringify(self.cta_order)}_MFMA"
+
 
-        return f"MFMA_{self.version}_{stringify(self.instr_shape)}_{self.transposed}_{stringify(self.warps_per_cta)}_{stringify(self.tiles_per_warp)}_{self.elem_type}_{stringify(self.ctas_per_cga)}_{stringify(self.cta_split_num)}_{stringify(self.cta_order)}_MFMA"
+@dataclass(frozen=True)
+class PaddedSharedLayout(DistributedLayout):
```

Comment:
This is a shared layout variant so base class should be `SharedLayout`? Also it should be `@dataclass(frozen=True, eq=True)`.